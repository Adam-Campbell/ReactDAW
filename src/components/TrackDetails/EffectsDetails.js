import React from 'react';
import PropTypes from 'prop-types';
import EffectDetailItem from './EffectDetailItem';
import { effectTypes } from '../../constants';

const EffectsDetails = props => (
    <div>
        <h2 className="track-details__heading">Effects</h2> 
        {props.effects.map(effect => (
            <EffectDetailItem 
                key={effect.id}
                effectType={effect.type}
                effectId={effect.id}
                handleOpen={props.handleOpen}
                handleRemove={props.handleRemove}
            />
        ))} 
        <div>
            <label 
                htmlFor="effect-select" 
                className="track-details__effect-select-label"
            >Add effect:</label>
            <select 
                id="effect-select" 
                className="track-details__effect-select"
                value={props.effectTypeToAdd}
                onChange={props.handleChange}
            >
                <option 
                    value={effectTypes.chorus}
                    className="track-details__effect-select-option"
                >Chorus</option>
                <option 
                    value={effectTypes.compressor}
                    className="track-details__effect-select-option"
                >Compressor</option>
                <option 
                    value={effectTypes.distortion}
                    className="track-details__effect-select-option"
                >Distortion</option>
                <option 
                    value={effectTypes.eq3}
                    className="track-details__effect-select-option"
                >EQ3</option>
                <option 
                    value={effectTypes.feedbackDelay}
                    className="track-details__effect-select-option"
                >Feedback Delay</option>
                <option 
                    value={effectTypes.filter}
                    className="track-details__effect-select-option"
                >Filter</option>
                <option 
                    value={effectTypes.freeverb}
                    className="track-details__effect-select-option"
                >Freeverb</option>
                <option 
                    value={effectTypes.gate}
                    className="track-details__effect-select-option"
                >Gate</option>
                <option 
                    value={effectTypes.jcReverb}
                    className="track-details__effect-select-option"
                >JC Reverb</option>
                <option 
                    value={effectTypes.limiter}
                    className="track-details__effect-select-option"
                >Limiter</option>
                <option 
                    value={effectTypes.phaser}
                    className="track-details__effect-select-option"
                >Phaser</option>
                <option 
                    value={effectTypes.tremolo}
                    className="track-details__effect-select-option"
                >Tremolo</option>
                <option 
                    value={effectTypes.vibrato}
                    className="track-details__effect-select-option"
                >Vibrato</option>
            </select>
            <button
                className="track-details__effect-button"
                onClick={() => props.handleAdd(props.effectTypeToAdd)}
            >Add</button>
        </div>
    </div>
);

EffectsDetails.propTypes = {
    handleOpen: PropTypes.func.isRequired,
    handleRemove: PropTypes.func.isRequired,
    handleAdd: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    effects: PropTypes.arrayOf(PropTypes.object).isRequired,
    effectTypeToAdd: PropTypes.string.isRequired
};

export default EffectsDetails;