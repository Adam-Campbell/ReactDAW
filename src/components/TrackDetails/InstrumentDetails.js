import React from 'react';
import PropTypes from 'prop-types';
import { synthTypes } from '../../constants';

const InstrumentDetails = props => (
    <div className="track-details__instrument-info-container">
        <h2 className="track-details__heading">Instrument</h2>
        <label 
            htmlFor="instrument-select" 
            className="track-details__instrument-select-label"
        >Current instrument</label>
        <select 
            id="instrument-select" 
            className="track-details__instrument-select"
            value={props.instrumentType}
            onChange={props.handleChange}
        >
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.default}
            >Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.am}
            >AM Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.fm}
            >FM Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.duo}
            >Duo Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.mono}
            >Mono</option>
        </select>
        <button 
            className="track-details__instrument-settings-button"
            onClick={props.handleOpen}
        >Edit Settings</button>
    </div>
);

InstrumentDetails.propTypes = {
    instrumentType: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleOpen: PropTypes.func.isRequired
};

export default InstrumentDetails;