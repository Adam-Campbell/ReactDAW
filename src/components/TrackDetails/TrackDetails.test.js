import React from 'react';
import { TrackDetails } from './';
import InstrumentDetails from './InstrumentDetails';
import EffectsDetails from './EffectsDetails';
import { shallow } from 'enzyme';

const channelsState = [
    {
        id: '3821569482884337',
        name: 'Channel 1',
        color: '#d86597',
        instrumentId: '5870129581156841',
        effectIds: [
            '3666421382061490'
        ],
        sectionIds: []
    }
];

const instrumentsState = {
    '5870129581156841': {
        id: '5870129581156841',
        channelId: '3821569482884337',
        type: 'synth',
        synthData: {
            envelope: {
                attack: 0.005,
                attackCurve: 'linear',
                decay: 0.1,
                release: 1,
                releaseCurve: 'exponential',
                sustain: 0.3
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'triangle',
                volume: 0
            },
            portamento: 0,
            volume: 0
        }
    }
};

const effectsState = {
    '3666421382061490': {
        id: '3666421382061490',
        type: 'chorus',
        channelId: '3821569482884337',
        effectData: {
            delayTime: 3.5,
            depth: 0.7,
            frequency: 1.5,
            spread: 180,
            type: 'sine',
            wet: 1
        }
    }
};

test('renders correctly', () => {
    const component = shallow(
        <TrackDetails 
            trackId="3821569482884337"
            channels={channelsState}
            instruments={instrumentsState}
            effects={effectsState}
            addInstrument={jest.fn()}
            removeInstrument={jest.fn()}
            openWindow={jest.fn()}
            addEffect={jest.fn()}
            removeEffect={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
    expect(component.find(InstrumentDetails)).toHaveLength(1);
    expect(component.find(EffectsDetails)).toHaveLength(1);
});
