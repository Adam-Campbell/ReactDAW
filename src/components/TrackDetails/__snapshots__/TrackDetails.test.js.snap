// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="track-details__container"
>
  <InstrumentDetails
    handleChange={[Function]}
    handleOpen={[Function]}
    instrumentType="synth"
  />
  <EffectsDetails
    effectTypeToAdd="chorus"
    effects={
      Array [
        Object {
          "channelId": "3821569482884337",
          "effectData": Object {
            "delayTime": 3.5,
            "depth": 0.7,
            "frequency": 1.5,
            "spread": 180,
            "type": "sine",
            "wet": 1,
          },
          "id": "3666421382061490",
          "type": "chorus",
        },
      ]
    }
    handleAdd={[Function]}
    handleChange={[Function]}
    handleOpen={[Function]}
    handleRemove={[Function]}
  />
</div>
`;
